### PR TITLE
Add module `muse/call` 

### LINKED ISSUE #5621: new module: MuSE
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### Additional Info
[MuSE is a somatic mutation calling tool](https://github.com/wwylab/MuSE/tree/master) for whole-genome sequencing (WGS) and whole-exome sequencing (WES) data from heterogeneous tumor samples. The latest version [was published recently.](https://genome.cshlp.org/content/early/2024/04/05/gr.278456.123.short)

### PR DESCRIPTION
**Pull request by famosab**
## PR checklist

Closes #5621 

- [ ] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

The Docker image still needs to be pushed to the nf-core registry, then I can update the paths accordingly. Also relates to #5629

### CODE REVIEW

**Comment by SPPearce** [comment]
Could this tool be added to BioConda? Then the other containers would be generated automatically

**Comment by famosab** [comment]
I opened a PR: https://github.com/bioconda/bioconda-recipes/pull/47957

If that works, I will change the files in this PR accordingly

File: modules/nf-core/muse/call/main.nf [hunk 1]
```diff
@@ -0,0 +1,52 @@
+process MUSE_CALL {
+    tag "$meta.id"
+    label 'process_high'
```
Comment (SPPearce) [inline modules/nf-core/muse/call/main.nf:3]: Does this need process_high?

File: modules/nf-core/muse/call/meta.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/meta-schema.json
+name: "muse_call"
+description: pre-filtering and calculating position-specific summary statistics using
+  the Markov substitution model
+keywords:
+  - genomics
+  - wgs
+  - wxs
+  - vcf
```
Comment (SPPearce) [inline modules/nf-core/muse/call/meta.yml:9]: These seem very generic to me.

File: modules/nf-core/muse/call/meta.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      licence: ["https://github.com/danielfan/MuSE/blob/master/LICENSE"]
+      identifier: ""
+
+input:
+  - - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. `[ id:'sample1' ]`
+    - tumor_bam:
```
Comment (SPPearce) [inline modules/nf-core/muse/call/meta.yml:27]: Does the tool use the indexes of the sorted bam files if they are present? If so, they should be passed in too.

File: modules/nf-core/muse/call/meta.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        description: Sorted matched normal BAM file
+        pattern: "*.bam"
+  - - meta2:
+        type: map
+        description: |
+          Groovy Map containing reference information.
+          e.g. `[ id:'test' ]`
+    - reference:
+        type: file
+        description: faidx indexed reference genome file
```
Comment (SPPearce) [inline modules/nf-core/muse/call/meta.yml:42]: An indexed file? Does that mean it wants the index to be present alongside?

**Review by SPPearce — COMMENTED** [review]
A few comments.

File: modules/nf-core/muse/call/main.nf (hunk 1 above)
Comment (famosab) [reply modules/nf-core/muse/call/main.nf:3]: probably not

File: modules/nf-core/muse/call/meta.yml (hunk 3 above)
Comment (famosab) [reply modules/nf-core/muse/call/meta.yml:28]: i am not sure, from the tool description it seems that it takes care of this inside of the tool (tabix is one of the dependencies)

File: modules/nf-core/muse/call/meta.yml (hunk 4 above)
Comment (famosab) [reply modules/nf-core/muse/call/meta.yml:43]: I just copied that from the module description but to me it seems that is also happy with just the fasta file

File: modules/nf-core/muse/call/meta.yml (hunk 3 above)
Comment (SPPearce) [reply modules/nf-core/muse/call/meta.yml:28]: Can you ask the authors please? As often the tools assume that the index is next to the bam file, which will generally work fine with symlinks on a HPC system but isn't true for cloud execution.

**Review by SPPearce — APPROVED** [review]
It would be good to check about the indexes (bam and fasta), but otherwise I'm happy.

**Comment by famosab** [comment]
I asked here: https://github.com/wwylab/MuSE/issues/31 
Will wait for their reply and merge afterwards :)

**Comment by famosab** [comment]
@SPPearce I added the `.bai` files alongside the `.bam` building ontop of the answer I got from the developers. Do you think its ok like this?

**Review by SPPearce — APPROVED** [review]
Ok, looks good to me now.

**Pull request merged**